lib/perfdata/influxdbwriter.cpp

```diff
@@ -615,3 +631,12 @@ void InfluxdbWriter::ValidateServiceTemplate(const Lazy<Dictionary::Ptr>& lvalue
 	}
 }
 
+int InfluxdbWriter::GetQueryCount(RingBuffer::SizeType span)
+{
+	return m_WorkQueue.GetTaskCount(span);
+}
+
+int InfluxdbWriter::GetPendingQueryCount()
+{
+	return m_WorkQueue.GetLength();
+}
```

Comment:
Are you sure the numbers you return here are exactly the ones you want?

Reply:
How do you find out how many pending WorkQueue are left?

Reply:
What do you need the pending _WorkQueue items_ for? Have a look how the IDO does that.

Reply:
Didn't I do the same thing as this?

https://github.com/Icinga/icinga2/blob/1b9f161aea8861e31c2744c7493d5e1c984c951e/lib/db_ido_pgsql/idopgsqlconnection.cpp#L1027-L1030

Reply:
Have a look how _MySQL_ calcs [queries/time](https://github.com/Icinga/icinga2/blob/91ffa89dad41fce6d9feee1209bb8273dd523cf0/lib/db_ido/dbconnection.cpp#L521-L533) and [pending queries](https://github.com/Icinga/icinga2/blob/91ffa89dad41fce6d9feee1209bb8273dd523cf0/lib/db_ido/dbconnection.cpp#L550-L560).

Reply:
However, the ``InfluxdbWriter`` does not have any queries that can be  in/decreased. I just wanted to find out what you can normally find in the Icinga2.log file.

```
[2021-01-22 14:01:38 +0100] information/WorkQueue: #7 (InfluxdbWriter, influxdb) items: 0, rate:  0/s (0/min 0/5min 0/15min);
```

Reply:
No queries, but perfdata to be written.

Reply:
Actually https://github.com/Icinga/icinga2/pull/8608#discussion_r562586995 still applies to GetQueryCount.